src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +73,147 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
+  // close Carousel when user press esc
+  useEffect(() => {
+    const keydownHandler = (evnt: KeyboardEvent) => {
+      evnt.key === "Escape" && setShowCarousel(false);
+    };
+    document.addEventListener("keydown", keydownHandler);
+    return () => document.removeEventListener("keydown", keydownHandler);
+  });
 
+  const useCarousel = (index: number) => {
+    showModal && setShowCarousel(true);
+    setSelectedImgIndex(index);
+  };
+  const closeCarousel = () => {
+    setShowCarousel(false);
+  };
   return (
-    <ImageGrid
-      height={imagesGridHeight}
-      width={imagesGridWidth}
-      numberOfImgs={numberOfImgs}
-      row={rowCol.row}
-      col={rowCol.col}
+    <div
+      className="container"
+      onClick={(evnt) => {
```

Comment:
could you also explain what you're trying to achieve with this function?

Reply:
Hi Alex, The purpose of this funciton is: 

when user click outside of the carousel box, ( mean that they click to the div have "container" class )

Reply:
ok, thanks. On that, try checking out this and look for similar related articles:
https://tekolio.com/how-to-detect-a-click-outside-of-a-react-component-using-hooks/#:~:text=To%20detect%20a%20click%20outside%20using%20hooks%2C%20you%20can%20utilize,the%20click%20outside%20behavior%20accordingly.

The idea is using `useRef` hook to check that element clicked is not the carousel element

Reply:
@Melody-Le have you had the time to check the usage of `useRef`?

Reply:
Hi @sashabaranova  . Yes. I did read the article and implementing it in files. 
It show here: [https://github.com/Melody-Le/fb-images-layout-storybook-component/blob/carousel/src/components/ImageGrid/ImageGridComponent.tsx](url)
Line 108 .
You see any issues?